Sources/SwiftDocCUtilities/Action/Actions/Init/InitAction.swift

```diff
@@ -0,0 +1,119 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021-2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import SwiftDocC
+
+/// An action that generates an article-only documentation catalog from a template seed.
+public struct InitAction: Action {
+    
+    enum Error: DescribedError {
+        case catalogAlreadyExists, cannotCreate
+        var errorDescription: String {
+            switch self {
+                case .catalogAlreadyExists: return "A documentation catalog with the same name already exists in the output directory."
+                case .cannotCreate: return "Unable to generate the documentation catalog."
+            }
+        }
+    }
+    
+    private let catalogOutputPath: String
+    private let documentationTitle: String
+    private let catalogTemplate: CatalogTemplateKind
+    private let includeTutorial: Bool
+    private let diagnosticEngine: DiagnosticEngine = DiagnosticEngine(treatWarningsAsErrors: false)
+    
+    /// Creates a new Init action from the given parameters
+    ///
+    /// - Parameters:
+    ///     - catalogOutputDirectory: The path to the directory where the catalog will be stored.
+    ///     - documentationTitle: The title of the technology root.
+    ///     - catalogTemplate: The template used to initialize the catalog.
+    ///     - includeTutotial: The boolean that indicates if the tutorial template should be added to the otput catalog.
+    public init(
+        catalogOutputDirectory: String,
+        documentationTitle: String,
+        catalogTemplate: CatalogTemplateKind,
+        includeTutorial: Bool
+    ) throws {
+        self.documentationTitle = documentationTitle
+        self.includeTutorial = includeTutorial
+        self.catalogTemplate = catalogTemplate
+        catalogOutputPath = "\(catalogOutputDirectory)/\(documentationTitle).docc"
+    }
+    
+    /// Generates a documentation catalog from a catalog template
+    ///
+    /// - Parameter logHandle: The file handle that the convert and preview actions will print debug messages to.
+    public mutating func perform(logHandle: SwiftDocC.LogHandle) throws -> ActionResult {
+        try initAction()
+    }
+                            
+    func initAction() throws -> ActionResult {
+        
+        let catalogOutputURL = URL(fileURLWithPath: catalogOutputPath)
+        let fileManager = FileManager.default
+        var logHandle: LogHandle = .standardError
+        var initProblems: [Problem] = []
+        diagnosticEngine.filterLevel = .warning
+        diagnosticEngine.add(DiagnosticConsoleWriter(formattingOptions: []))
+        
+        defer {
+            diagnosticEngine.emit(initProblems)
```

Comment:
When I tested this I found that this call did not write anything to stdout... the diagnostics were not displayed in the terminal. I'm not sure what's wrong, but in the debugger this code in `DiagnosticsEngine` is not working:

```
        workQueue.async { [weak self] in
            // If the engine isn't around then return early
            guard let self = self else { return }
            for consumer in self.consumers.sync({ $0.values }) {
                consumer.receive(filteredProblems)
            }
        }
```

Wild guess is that the overall docc process ends before the thread started here by `workQueue.async` gets a chance to run? Not sure.

Reply:
Thanks for pointing this out Pat! to get the errors printed to the user I had to invoke both methods
```
defer {
      diagnosticEngine.emit(initProblems)
      diagnosticEngine.flush()
  }
```
Change introduced in https://github.com/apple/swift-docc/pull/719/commits/cb1d6fc580bb1d99888edf8a6a3121338ed76d7c